contracts/RealmDiamond/facets/ParcelRolesRegistryFacet.sol

```diff
@@ -30,17 +30,43 @@ contract ParcelRolesRegistryFacet is Modifiers, IERC7432 {
   function grantRole(Role calldata _role) external override onlyValidRole(_role.roleId) onlyRealm(_role.tokenAddress) {
     require(_role.expirationDate > block.timestamp, "ParcelRolesRegistryFacet: expiration date must be in the future");
 
-    // Deposit NFT if necessary and get the original owner
     address _originalOwner = _depositNft(_role.tokenAddress, _role.tokenId);
 
-    // Get role data from storage
     RoleData storage _roleData = s.erc7432_roles[_role.tokenAddress][_role.tokenId][_role.roleId];
 
-    // Check if the role is either revocable or expired
     require(_roleData.revocable || _roleData.expirationDate < block.timestamp, "ParcelRolesRegistryFacet: role must be expired or revocable");
 
     s.erc7432_roles[_role.tokenAddress][_role.tokenId][_role.roleId] = RoleData(_role.recipient, _role.expirationDate, _role.revocable, _role.data);
```

Comment:
I wanna to double check this with you (maybe on a call). But I'm thinking on not storing `data` on storage, and on the view function, we encode the profitShare info (which should be in storage)